internal/pkg/cli/errors.go

```diff
@@ -163,3 +163,25 @@ func (e *errPipelineDependsOnEnv) RecommendActions() string {
 or run %s to delete the pipeline before running %s to delete the environment`,
 		e.pipeline, e.env, color.HighlightCode(fmt.Sprintf("copilot pipeline delete -n %s", e.pipeline)), color.HighlightCode(fmt.Sprintf("copilot env delete -n %s", e.env)))
 }
+
+type errTaskRoleRetrievalFailed struct {
+	chainErrs []error
+}
+
+func (e *errTaskRoleRetrievalFailed) Error() string {
+	return errors.Join(e.chainErrs...).Error()
+}
```

Comment:
could this just be 
```suggestion
type errTaskRoleRetrievalFailed error
```
and then created down below like:
```
errTaskRoleRetrievalFailed(errors.Join(errs...))
```

Reply:
I'm not entirely certain what you're suggesting but I don't think this is useful. Since `error` is an interface this would just mean defining a new interface, more verbosely like this: `type errTaskRoleRetrievalFailed interface { error }` correct? How would I work with this to simplify the code here.

Reply:
similar to what we do here! https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/aws/ecs/service.go#L25 basically - the only thing we're doing is adding a function the type, otherwise it's still just an error. just a small thing, i don't feel super strongly either way 😊